lib/services/devices/deviceService.js

```diff
@@ -403,7 +403,15 @@ function updateRegisterDevice(deviceObj, callback) {
  */
 function listDevices(service, subservice, limit, offset, callback) {
     if (!callback) {
-        callback = limit;
+        if (service && subservice && limit) {
+            callback = limit;
+        } else if (service) {
+            callback = service;
+            service = null;
+            subservice = null;
+        } else {
+            logger.fatal(context, 'Couldn\'t find callback in listDevices() call.');
+        }
```

Comment:
Kind of confusing to me these assignments to the `callback` parameter... :) Taking into consideration it ends up being passed to the listDevices() function at https://github.com/telefonicaid/iotagent-node-lib/blob/bb39fff6494df14437c56b85e1b7cf8e7b8f8039/lib/services/devices/deviceRegistryMongoDB.js#L107 Maybe it is because it is Tuesday after a long weekend... :)